sdk/core/Azure.Core/src/Pipeline/Policies/RequestActivityPolicy.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using System.Globalization;
+using System.Threading.Tasks;
+
+namespace Azure.Core.Pipeline.Policies
+{
+    internal class RequestActivityPolicy : HttpPipelinePolicy
+    {
+        public static RequestActivityPolicy Shared { get; } = new RequestActivityPolicy();
+
+        private static readonly DiagnosticListener s_diagnosticSource = new DiagnosticListener("Azure.Pipeline");
+
+        public override Task ProcessAsync(HttpPipelineMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
+        {
+            return ProcessAsync(message, pipeline, true);
+        }
+
+        public override void Process(HttpPipelineMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
+        {
+            ProcessAsync(message, pipeline, false).EnsureCompleted();
+        }
+
+        private static async Task ProcessAsync(HttpPipelineMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline, bool isAsync)
+        {
+            if (!s_diagnosticSource.IsEnabled())
+            {
+                if (isAsync)
+                {
+                    await ProcessNextAsync(message, pipeline).ConfigureAwait(false);
+                }
+                else
+                {
+                    ProcessNext(message, pipeline);
+                }
+
+                return;
+            }
+
+            var activity = new Activity("Azure.Core.Http.Request");
+            activity.AddTag("http.method", message.Request.Method.Method);
```

Comment:
I suggest passing HttpRequest and response objects - and avoid using tags for http pipeline.

If you have a strong need to use them to avoid more work in bridge, check if activity is sampled in first, before adding them.

We should talk more about how to support it better: Activity.Recorded, it comes with next DiagnosticSource 4.6.0-preview), it is propagated from parent and it could be changed in Start events.

Reply:
There is some friction with updating to DiagnosticSource 4.6.0-preview because of the central version management in the repo. I'll file an issue to follow up.

Reply:
> If you have a strong need to use them to avoid more work in bridge, check if activity is sampled in first, before adding them.

Isn't checking for `s_diagnosticSource.IsEnabled(activity.OperationName, message.Request)` enough?

Reply:
if IsEnabled is false, then Activity is not created at all - nothing gets tracked or propagated. Activity.Recorded = false means activity is created, and propagated, but augmenting it with anything does not make sense.

You see why we need OpenTelemetry in general - there are some many caveats around Activity...

Reply:
> activity is created, and propagated, but augmenting it with anything does not make sense.

What is the point of this activity then? It carries no information except ID and operation name.